tests/TaskEventTest: AssertionError: ev_start not called

I am experiencing weird behavior of tests when I try to run just subset of tests. For example
```sh
nosetests -v --all-modules tests/TaskEventTest.py
```
all tests pass but
```sh
nosetests -v --all-modules tests/StreamWorkerTest.py tests/TaskEventTest.py
```
do not. It fails with `ev_start` related errors (see https://github.com/arkamar/clustershell/runs/8268340929?check_suite_focus=true#step:9:40):
```
======================================================================
FAIL: test ev_pickup event with fanout
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/clustershell/clustershell/tests/TaskEventTest.py", line 516, in test_ev_pickup_fanout
    eh.do_asserts_noread_notimeout()
  File "/home/runner/work/clustershell/clustershell/tests/TaskEventTest.py", line 40, in do_asserts_noread_notimeout
    assert self.did_start, "ev_start not called"
AssertionError: ev_start not called

======================================================================
FAIL: test ev_pickup event with fanout (legacy)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/clustershell/clustershell/tests/TaskEventTest.py", line 495, in test_ev_pickup_fanout_legacy
    eh.do_asserts_noread_notimeout()
  File "/home/runner/work/clustershell/clustershell/tests/TaskEventTest.py", line 40, in do_asserts_noread_notimeout
    assert self.did_start, "ev_start not called"
AssertionError: ev_start not called
```
Interestingly, if I switch order of files
```sh
nosetests -v --all-modules tests/TaskEventTest.py tests/StreamWorkerTest.py
```
everything works again.

--- a/tests/StreamWorkerTest.py
+++ b/tests/StreamWorkerTest.py
@@ -21,7 +21,13 @@ class StreamTest(unittest.TestCase):
         """test empty StreamWorker"""
         # that makes no sense but well...
         # handler=None is supported by base Worker class
-        self.run_worker(StreamWorker(handler=None))
+        worker = StreamWorker(handler=None)
+        self.run_worker(worker)
+        # GH Issue #488:
+        # An unconfigured engine client does not abort by itself...
+        worker.abort()
+        # Check that we are in a clean state now
+        self.assertEqual(len(task_self()._engine._clients), 0)
 
     def test_002_pipe_readers(self):
         """test StreamWorker bound to several pipe readers"""

